Ops/Ops-lessons-2022/AWS.md

```diff
@@ -71,16 +71,67 @@ Documentation:
 ### Cloudwatch
 
 ### EventBridge
+- Abstracted version of cloudwatch events
+- Allows users to create rules that initiate events that trigger a target it has set in that rule
+- Eventbus is a collection of rules tha triggers targets
+- In eventbus an event has an rule and a target
+  - Whenever the rule is matched, the target is initiated
+
+SecurEd Use Cases:
+- Trigger all scheduled tasks in ECS
+- Trigger SNS and bundling service
+
+[AWS CloudTrail](https://docs.aws.amazon.com/eventbridge/latest/userguide/eb-what-is.html)
 
 ### CloudTrail
+- Tracks AWS API calls
+- Logs data in S3 buckets
+- CloudTrail insights logs unusual API activity
+- CloudTrail event history stores all API activity within the last 90 days
+  - Who called the action
+  - What service was called
+  - When the user called the action
+  - Where the user called it from
+
+SecurEd Use Cases:
+- Debugging user permissions
+
+[AWS CloudTrail](https://docs.aws.amazon.com/awscloudtrail/latest/userguide/cloudtrail-user-guide.html)
 
 ### Simple Notification Service (SNS)
 
 ### Secrets Manager
+- Used to hold secrets for an AWS account
+
+SecurEd Use Cases:
+- Store IAM key pairs
+- Store Admin credentials
+- Store secrets for other non-AWS services
+- A middle man for key-rotation-service to move credentials from IAM to secrets manager then to the service that the key needs to be rotated on
+- Store any sensitive data used for developmental or production purposes
+
+Documentation:
+- [AWS Secrets Manager](https://aws.amazon.com/secrets-manager/)
 
 ### Simple Storage Service (S3)
 
 ### Virtual Private Cloud (VPC)
+- Heart of networking at AWS
+- Enables launching of a complete virtual network and mimics a data center
```

Comment:
How does it mimic a data center?

Reply:
Based on the information found [here](https://www.paloaltonetworks.com/cyberpedia/what-is-a-data-center) and looking at the primary elements of a datacenter, VPC is practically a virtual data center. 
* The facility is the virtual interface on the web we use to interact with our virtual cloud environments
* We handle the cases of core components like routing, firewall, storage systems (not used in our case), servers (EC2, ECS, etc...), and security elements (security groups, network acls, etc...) all through VPC
* Support infrastructure and operations staff is handled by AWS internally so we never have to worry about that.

Reply:
Huh neat!  Thank you!